test/doc/end-to-end-test-config.md

```diff
@@ -0,0 +1,45 @@
+| `relayerImage` | * | Relayer image to be used. Used when running PriorityQueue tests, ignored otherwise.|
+| `rootCaCertificatePath` | * | Full path to a root certificate given to leaf test devices so they can verify the authenticity of Edge Hub during the TLS handshake. Used when running the test 'TransparentGateway', ignored otherwise. |
+| `rootCaPrivateKeyPath` | * | Full path to a file containing the private key associated with `rootCaCertificatePath`. Used when running the test 'TransparentGateway', ignored otherwise. |
+| `setupTimeoutMinutes` || The maximum amount of time, in minutes, test setup should take. This includes setup for all tests, for the tests in a fixture, or for a single test. If this time is exceeded, the associated test(s) will fail with a timeout error. If not given, the default value is `5`. |
+| `teardownTimeoutMinutes` || The maximum amount of time, in minutes, test teardown should take. This includes teardown for all tests, for the tests in a fixture, or for a single test. If this time is exceeded, the associated test(s) will fail with a timeout error. If not given, the default value is `2`. |
+| `tempFilterFuncImage` | * | Azure temperature filter function to be used. Used when running the test 'TempFilterFunc', ignored otherwise.|
+| `tempFilterImage` | * | Docker image to pull/use for the temperature filter module. Used when running the test 'TempFilter', ignored otherwise.|
+| `tempSensorImage` || Docker image to pull/use for the temperature sensor module (see the test 'TempSensor'). If not given, `mcr.microsoft.com/azureiotedge-simulated-temperature-sensor:1.0` is used.|
+| `numberLoggerImage` || Docker image to pull/use for the Edge agent direct method tests. Used to generate predictable logs. |
+| `testResultCoordinatorImage` | * | TestResultCoordinator image to be used. Used when running PriorityQueue or GenericMqtt tests, ignored otherwise.|
+| `genericMqttTesterImage` | * | GenericMqttTester image to be used. Used when running GenericMqtt tests, ignored otherwise.
+| `testTimeoutMinutes` || The maximum amount of time, in minutes, a single test should take. If this time is exceeded, the associated test will fail with a timeout error. If not given, the default value is `5`. |
+| `verbose` || Boolean value indicating whether to output more verbose logging information to standard output during a test run. If not given, the default is `false`. |
+
+# [Environment Variables](#environment-variables)
+
+| Name | Required | Description |
+|------|----------|-------------|
+| `E2E_DPS_GROUP_KEY` | ✔ | The symmetric key of the DPS [enrollment group](https://docs.microsoft.com/en-us/azure/iot-dps/concepts-service#enrollment-group) to use. Used when running any DPS tests, ignored otherwise. |
```

Comment:
`E2E_DPS_GROUP_KEY` should have an '*' not a '✔' in the Required column, since it's only required for DPS tests, not all tests.

Reply:
...although I see several mistakes in the Required column, and probably none of them are from you. :) Perhaps we need to make a run the the Required column as a separate effort.

Reply:
Wasn't sure what the semantic difference was between the ✔ and the *.  I'll document that in the PR i.e, ✔ means required and * means conditionally required. 

Also, I think we should take this opportunity to clean up the Required column. I did find that to be pretty confusing as I was setting up my environment.

Reply:
Yep, agreed...anything we can do to make this easier to read and understand is a win.

Reply:
Changed the ✔and added a blurb about the difference between * and ✔